File: template/test_solution.py
import pytest
from solution import Solution1, Solution2

SOLUTIONS = [Solution1(), Solution2()]

@pytest.mark.parametrize("sol", SOLUTIONS)
class TestSolution:
    def test_case1(self, sol):
        pass  # TODO: replace with real test

    def test_case2(self, sol):
        pass  # TODO: replace with real test

File: template/test_solution.cpp
#include <gtest/gtest.h>
#include "solution.cpp"

template <typename T>
class SolutionTest : public testing::Test {
protected:
    T sol;
};

using Implementations = testing::Types<Solution1, Solution2>;
TYPED_TEST_SUITE(SolutionTest, Implementations);

TYPED_TEST(SolutionTest, Case1) {
    // TODO: replace with real test
}
